File: probability-example-main/probability-example-PythonVersion/python/Menu.py
import GameRunner
import sys
import fileinput

GAMES = ['Coin Flip', 'Dice Roll', 'Card Game']

def print_menu(score, points):
    print("You can play:")
    for i, game in enumerate(GAMES):
        if (score >= points[i]):
            print(f"{i+1}. {game} ({points[i]} points)")
    print("0. Quit while you are ahead!")
    print("Please enter the number for your choice: ")

def main():
    player_points = 10
    runner = GameRunner.GameRunner()

    print("Welcome to the Probability Game!")
    print("Please enter your name:")
    player_name = input()

    # Attempt to load the player score
    filename = "scores.txt"
    try:
        print("Trying to open the file")
        with open(filename) as fin:
            for line in fin:
                parts = line.split('||')
                if len(parts) >= 2 and parts[0] == player_name:
                    player_points = int(parts[1]) # This could throw an exception, but assume parse works
                    break;
    except:
        print("Failed to open file.  Making a new one.")
        f = open(filename, 'a')
        f.close();

    choice = -1;
    while (choice != 0 and player_points > 0 and player_points < 100):
        points = [runner.coin_points, runner.dice_points, runner.card_points]
        print(f'\nYou have {player_points} points.')
        print_menu(player_points, points)
        try:
            choice = int(input())
            if choice < 0 or choice > len(points) or player_points < points[choice-1]:
                raise ValueError
        except:
            print("Please enter a valid choice")
            choice = -1
            continue
        if choice == 0:
            print(f"You quit with {player_points} points.")
        elif choice == 1:
            print("\nCoin Game")
            player_points = runner.play_coin_game(player_points)
        elif choice == 2:
            print("\nDice Game")
            player_points = runner.play_dice_game(player_points)
        elif choice == 3:
            print("\nCard Game")
            player_points = runner.play_card_game(player_points)
    
    if player_points == 0:
        print("Oh no! You lost it all!  Better luck next time!")
    elif  player_points >= 100:
        print("Wow!  You broke the bank!  Congratulations!")


    # Ask if the player wants to save their score to the file
    print("Would you like to save your name and score?  Enter Y or N: ")
    choice = GameRunner.GameRunner.get_y_or_n()
    updated = False
    if (choice == 'Y'):
        with open(filename, 'r') as f:
            lines = f.readlines()
        with open(filename, 'w') as f:
            for line in lines:
                parts = line.split('||')
                if len(parts) >= 2 and parts[0] == player_name:
                    f.write(f"{player_name}||{player_points}\n")
                    updated = True
                else:
                    f.write(line)
            if not updated:
                f.write(f"{player_name}||{player_points}\n")

if __name__ == "__main__":
    main()

File: probability-example-main/probability-example-PythonVersion/csharp/Menu.cs
using System;

class Menu
{
    public static string GetYN(){
        string choice = Console.ReadLine().ToUpper();
        while (choice != "Y" && choice != "N"){
            Console.Write("Please enter Y or N: ");
            choice = Console.ReadLine().ToUpper();
        } 
        return choice;
    }
    public static void PrintMenu(int max){
        Console.WriteLine("You can play: ");
        //only print the games the player has enough points to play
        if(max >= 1){
            Console.WriteLine("1. Coin Flip (1 point)");
        }
        if(max >= 2){
            Console.WriteLine("2. Dice Roll (2 points)");
        }
        if(max >= 3){
            Console.WriteLine("3. Card Game (5 points)");
        }
        Console.WriteLine("0. Quit while you are ahead!");
        Console.WriteLine("\nPlease enter the number for you choice: ");
    }
    static int ReadInt(int min, int max){
        string readValue;
        int readInt;
        
        readValue = Console.ReadLine();
        readInt = int.Parse(readValue);
        while(readInt < min || readInt > max){
            Console.Write($"Please enter an integer between {min} and {max}. ");
            readValue = Console.ReadLine();
            readInt = int.Parse(readValue);
        }
        return readInt;
    }
    public static void Main(string[] args)
    {
        int playerPoints = 10;
        GameRunner gr = new GameRunner();
        string filename = "scores.txt";
        string[] lines = System.IO.File.ReadAllLines(filename);
        string saveLoad;
        int index = -1;

        Console.WriteLine("Welcome to the Probability Game!");
        Console.WriteLine("Please enter your name: ");
        string playerName = Console.ReadLine();
        
        //if there are saved games, check to see if this person has one
        if (lines.Length > 0){
            index = Array.FindIndex(lines, element => element.Contains(playerName));
            if(index >= 0){
                Console.WriteLine($"Welcome back, {playerName}!");
                Console.WriteLine("Would you like to load a previously saved game? Enter Y or N");
                saveLoad = GetYN();
                if(saveLoad == "Y"){
                    string[] parts = lines[index].Split("||");
                    playerPoints= int.Parse(parts[1]);
                }
            }
        }

        //determine which game to play
        int choice = 1;
        int max;
        while(choice!=0 && playerPoints > 0 && playerPoints < 100){
            //find the highest scoring game that can be played with the points available.
            if(playerPoints >= gr.cardPoints){
                max = 3;    
            }
            else if(playerPoints >= gr.dicePoints){
                max = 2;
            }
            else{
                max = 1;
            }
            //Print the menu and get the choice
            Console.WriteLine();
            Console.WriteLine($"You have {playerPoints} points.");
            PrintMenu(max);
            choice = ReadInt(0,max);

            switch(choice){
                case 0:
                    Console.WriteLine($"You quit with {playerPoints} points.");
                    if(playerPoints > 10){
                        Console.WriteLine("Nice job!");
                    }
                    else{
                        Console.WriteLine("Better luck next time!");
                    }
                    break;
                case 1:
                    Console.WriteLine("\nCoin Game");
                    playerPoints = gr.CoinGame(playerPoints);
                    break;
                case 2:
                    Console.WriteLine("Dice Game");
                    playerPoints = gr.DiceGame(playerPoints);
                    break;
                case 3:
                    Console.WriteLine("Card Game");
                    playerPoints = gr.CardGame(playerPoints);
                    break;
            }
        }
        if(playerPoints == 0){
            Console.WriteLine("Oh no! You lost it all! Better luck next time!");
        }
        else if(playerPoints >= 100){
            Console.WriteLine("Wow! You broke the bank! Congratulations!");
        }

        //Ask if want to save.
        Console.Write("Would you like to save your name and score? Enter Y or N: ");
        saveLoad = GetYN();
        if(saveLoad == "Y"){
            //if this player has played before, rewrite their info.
            if(index >= 0){
                lines[index] = $"{playerName}||{playerPoints}";
            }
            using (StreamWriter outputFile = new StreamWriter(filename)){
                //rewrite all the old info
                foreach(string line in lines){
                    outputFile.WriteLine(line);
                }
                //If this is a new player, write the information at the end
                if(index<0){
                    outputFile.WriteLine($"{playerName}||{playerPoints}");
                }
            }   
        }

    }//end Main
}